PTBED-235: APK Installation Shortcut

wil25632 opened the pull request:
Added a shortcut via `kvm-compose` for users to install an APK file onto an Android guest without going through for example `kvm-compose exec phone tool adb ...`.

wil25632 committed:
created a shortcut to install an apk file on android guests directly from the kvm-compose command line

wil25632 committed:
added the documentation of the new install-apk shortcut into docs/interfaces/exec/tool.rst

drrtdrrt commented on kvm-compose/kvm-compose-schemas/src/exec.rs:
```diff
@@ -84,6 +84,8 @@ pub enum TestbedTools {
     ADB(Command),
     FridaSetup,
+    /// Install an Android app from an APK file
+    InstallApk(Command),
```
Here `Command` provides just anything that the user provides, so we have no validation of what is given. Could you create something specific to this feature that documents itself like `ExecCmdFileTransfer` higher up in this file.

So the objective here is to give the user an option to give one argument which is the 'full path to apk' on the host. This can then be mapped to a `PathBuf` like in the given example, so that in the install function `install_apk` you can then 1) check that this exists and then 2) send that into the `adb` command to the emulator.

drrtdrrt reviewed (changes requested):
Looks good so far, just a quick change in the schema for the CLI options, plus a short validation for the file path input to make sure it exists.

Ideally, we also should have the corresponding GUI update to expose this command there. Message me on this when you pick this up, there are a couple of things to unpick here for that.

wil25632 committed:
added path validation and path canonicalisation for the apk file

wil25632 committed:
fixed the gui error of only capturing and sending the first character of the command when using tool on a guest

wil25632 committed:
implemented the feature on GUI as well; added the necessary values in toolOptions where the HTML is generated based off

wil25632 commented:
Completed the changes requested:
1. Added path canonicalisation for the APK file and validation to check if it exists before sending it to adb
2. While attempting to include this new shortcut to the GUI, discovered that only the first character of the arguments to the exec tools in the GUI were captured. Fixed that.
3. Implemented the new shortcut to the GUI with HTML elements introduced and everything, at the moment the argument to this shortcut requires the full path of the APK file in the GUI.

wil25632 committed:
Merge branch 'develop' into PTBED-235_apk-installation-shortcut

wil25632 committed:
removed the suppress_output parameter from install_apk function definition; added the check if the adb server is running at the start of install_apk function; removed the commented out InstallApk(Command); removed bail and the return of Err in install_apk to prevent duplicated error messages

wil25632 commented:
Completed the changes:

1. Removed the `suppress_output` parameter from `install_apk` function definition
2. Added the check if the `adb` server is running at the start of `install_apk` function
3. Removed the commented out `InstallApk(Command)`
4. Removed bail and the return of `Err` in `install_apk` to prevent duplicated error messages

wil25632 committed:
combined tracing::error! and return Err into a single bail! in install_apk()

drrtdrrt committed:
add path that gave error to canonicalise context error log

drrtdrrt reviewed (approved):
Tested CLI and GUI, all working. Added a context to canonicalize to prevent cryptic errors in the CLI if a path is wrong
